kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -193,13 +194,35 @@ def parse_llm_response(self, message: BaseMessage, language: str) -> _llm_respon
 
 def guess_language(code: str, filename: Optional[str] = None) -> str:
     try:
-        if filename is not None:
-            lexer = lexers.guess_lexer_for_filename(filename, code)
-            logger.debug(f"{filename} classified as {lexer.aliases[0]}")
+        all_lexers = lexers.get_all_lexers()
+        largest_rv_lexer = None
+        largest_rv = 0.0
+        for name, _, _, _ in all_lexers:
+            try:
+                lexer_found = cast(LexerMeta, lexers.get_lexer_by_name(name))
+            except ClassNotFound:
+                logger.log(DEBUG, "unable to find lexer class for name: %s", name)
```

Comment:
this is adding a lot of noise, but for some period of time, if we keep it at debug, I can deal with issues with this code, after that, we can move to trace